classes/constraints/isEmpty.php

```diff
@@ -3,27 +3,41 @@
 namespace mageekguy\atoum\phpunit\constraints;
 
 use mageekguy\atoum\asserters;
-use mageekguy\atoum\exceptions;
+use mageekguy\atoum\phpunit\constraint;
 use mageekguy\atoum\tools\variable\analyzer;
 
-class isEmpty extends count
+class isEmpty extends constraint
 {
+    private $analyzer;
+
     public function __construct($description = null, analyzer $analyzer = null)
     {
-        parent::__construct(0, $description, $analyzer);
+        $this->description = $description;
+        $this->analyzer = $analyzer ?: new analyzer();
     }
 
     protected function matches($actual)
     {
-        if ($this->analyzer->isString($actual)) {
-            $asserter = new asserters\phpString(null, $this->analyzer);
+        if ($actual instanceof \countable) {
+            $asserter = new asserters\sizeOf(null, $this->analyzer);
+            $asserter->setWith($actual)->isEqualTo(0, $this->description);
+        } else {
+            $asserter = new emptyAsserter(null, $this->analyzer);
             $asserter->setWith($actual)->isEmpty($this->description);
+        }
+    }
+}
+
+class emptyAsserter extends asserters\variable
```

Comment:
Please either:
 * extract this class to its own file
* include its code in the `isEmpty` constraint

Reply:
:+1:

Reply:
https://github.com/atoum/phpunit-extension/pull/22